tsl/src/chunk.c

```diff
@@ -632,6 +633,23 @@ merge_chunks_lock_upgrade_mode(void)
 	return MERGE_LOCK_ACCESS_EXCLUSIVE;
 }
 
+/*
+ * Use anonymous settings value to disable multidim merges due to a bug in the
+ * routing cache with non-aligned partitions/chunks.
+ */
+static bool
+merge_chunks_multidim_allowed(void)
+{
+	const char *multidim_allowed =
+		GetConfigOption("timescaledb.merge_chunks_multidim_allowed", true, false);
```

Comment:
Normal name format we use for these are "enable_...", so maybe stick with that naming convention. It also makes it easier to browse the code and find options, especially if you have "anonymous" options that are not visible through the SQL interface.

You might also want to use the macro we have for adding the extension namespace.

```suggestion
		GetConfigOption(MAKE_EXTOPTION("enable_merge_chunks_multidim"), true, false);
```